### ISSUE TITLE
Branch names are terminated at the first period

### ISSUE THREAD

**Issue by davekeane** [magicmonty/bash-git-prompt#178]
Branch names are terminated at the first period

e.g. I get the following prompt when on both 'development-3.1' and 'development-3.2' branches.
```
✔ 16:34 user@host:~/s/project [ development-3 | ✔  ] $
```

I've traced it to the behaviour of gitstatus.sh, and parsing of 
```
$ git status --porcelain --branch
## development-3.2.x...origin/development-3.2.x [behind 14]
```
, though haven't had a chance to consider an alternate implementation.

**Comment by ogr3** [comment]
That's a bug in the parsing of the branch line. @pedantic79

**Comment by ogr3** [comment]
There will be three different cases:
no upstream
```## development-3.2.x```
upstream clean
```## development-3.2.x...origin/development-3.2.x```
upstream with changes
```## development-3.2.x...origin/development-3.2.x [behind 14]```

either match from '## ' to the end of line or to the ellipsis

**Comment by easterwood** [comment]
What about ```branch="${line[0]%*\.\.\.*}"``` for getting the branch name?

**Comment by ogr3** [comment]
That should work as long as the branch name itself doesn't contain '...', but git doesn't allow that so yes, this seems to work.

**Comment by ogr3** [comment]
I don't know if the read construct is useful if '.' cannot be used as a separator.

Maybe just simplify to remove prefix and suffix from branch_line?

```
branch="${branch_line#\#\# }"
branch="${branch%\.\.\.*}"
```

Then you could do the same trick to extract the upstream branch if that is needed somewhere.

**Comment by easterwood** [comment]
The upstream is already available in the current master version.
```
....
else
  echo "${#line[@]}"
  echo "${line[3]}"
  if [[ "${#line[@]}" -eq 1 ]]; then
    remote="_NO_REMOTE_TRACKING_"
  else
    IFS="[,]" read -ra remote_line <<< "${line[3]}"
    upstream="${remote_line[0]}" // Only in my fork
    for rline in "${remote_line[@]}"; do
      if [[ "$rline" == *ahead* ]]; then
...
``` 

@pedantic79 implemented this. btw. thx.

So if  ```branch_line``` will remain the same than this should work as it does.

**Comment by ogr3** [comment]
The fix breaks the check for ahead/behind and local branch, since the original split the string on '.'

One should split the string on '...' and adapt the array indices in order to keep all functionality.

**Comment by ogr3** [comment]
This might work:

```
branch="${branch_line#\#\# }"
branch="${branch%\.\.\.*}"
after_branch="${branch_line#*\.\.\.}"
remote=

if [[ "$branch" == *"Initial commit on"* ]]; then
  IFS=" " read -ra branch_line <<< "$branch"
  branch="${branch_line[3]}"
  remote="_NO_REMOTE_TRACKING_"
elif [[ "$branch" == *"no branch"* ]]; then
  tag=$( git describe --exact-match )
  if [[ -n "$tag" ]]; then
    branch="$tag"
  else
    branch="_PREHASH_$( git rev-parse --short HEAD )"
  fi
else
#  if [[ "${#line[@]}" -eq 1 ]]; then
  if [[ "${after_branch}" == "${branch_line}" ]]; then
    remote="_NO_REMOTE_TRACKING_"
  else
    IFS="[,]" read -ra remote_line <<< "${after_branch}"
    for rline in "${remote_line[@]}"; do
      if [[ "$rline" == *ahead* ]]; then
        num_ahead=${rline:6}
        ahead="_AHEAD_${num_ahead}"
      fi
      if [[ "$rline" == *behind* ]]; then
        num_behind=${rline:7}
        behind="_BEHIND_${num_behind# }"
      fi
    done
    remote="${behind}${ahead}"
  fi
fi

```

**Comment by ogr3** [comment]
attempt at fix in #179 Needs some testing for all cases.

**Comment by davekeane** [comment]
@ogr3: I checked out your branch, and it 'works for bug case'
```
✔ 10:59 user@host:~/s/project [ development-3.2.x | ✔  ] $ git --version
git version 2.6.0
```
I've tested it against ahead/behind/noremote cases as well, and I've not encountered any errant behaviour there.

**Comment by ogr3** [comment]
Merged the proposal frpm @pedantic79. Will field test it today.

**Comment by ogr3** [comment]
Field tested during the day - seems to work as before, but with support for dots in branch names. @davekeane: Have you had a chance to test the latest version?

**Comment by davekeane** [comment]
Not the current dots-fix. I've checked it out, and will field test it a bit.

**Comment by magicmonty** [comment]
Should be fixed with #179
